belindas-closet-nestjs_4_66_add-unit-tests-for-archive-functionality

heosman opened the pull request:
Resolves #66 

This PR adds unit tests for archive functionality for the product controller and services modules.

Test results for products.controller.spec.ts:
<img width="563" alt="products controller spec ts results" src="https://github.com/SeattleColleges/belindas-closet-nestjs/assets/77591083/82d2333a-2972-45d2-8bf9-cf6872882ded">

Test results for products.service.spec.ts:
<img width="482" alt="products service spec ts results" src="https://github.com/SeattleColleges/belindas-closet-nestjs/assets/77591083/76bae245-a081-442f-85d9-daa94b8c359e">



Relates to #45

CourtneyHoppus commented:
@heosman it looks like when the delete functionality was changed in pull request #64 that the testing wasn't updated. What you have here doesn't test it fully. But I don't think this is your issue and should probably be addressed in it's own issue. @brinkbrink @taylorpapke @theGaryLarson @tinpham5614 @Robel-003 @Seiyaroo

CourtneyHoppus commented on src/products/services/products/products.service.spec.ts:
```diff
@@ -154,12 +155,24 @@ describe('ProductsService', () => {
-        { isHidden: true }
-      );
-      expect(result).toEqual(deletedProduct);
+      mockProduct = await service.delete(mockProduct.id);
```
You want to leave this as before, `const result = await service.delete(mockProduct.id)` as the mockProduct is already defined and we want to see what the service brings back.

CourtneyHoppus commented on src/products/services/products/products.service.spec.ts:
```diff
@@ -154,12 +155,24 @@ describe('ProductsService', () => {
-      );
-      expect(result).toEqual(deletedProduct);
+      mockProduct = await service.delete(mockProduct.id);
+      expect(mockProduct).toEqual(deletedProduct);
```
Here we still want to compare the `result` to the `deletedProduct`, so it should still be `expect(result).toEqual(deletedProduct);` to use the `result` from the service.

taylorpapke commented:
> @heosman it looks like when the delete functionality was changed in pull request #64 that the testing wasn't updated. What you have here doesn't test it fully. But I don't think this is your issue and should probably be addressed in it's own issue. @brinkbrink @taylorpapke @theGaryLarson @tinpham5614 @Robel-003 @Seiyaroo

I would prefer if this was done by @Robel-003 @tinpham5614 or @CourtneyHoppus as they have permission to use our issue naming convention for this and can create the correct issue to target this in a separate PR.

heosman commented:
> > @heosman it looks like when the delete functionality was changed in pull request #64 that the testing wasn't updated. What you have here doesn't test it fully. But I don't think this is your issue and should probably be addressed in it's own issue. @brinkbrink @taylorpapke @theGaryLarson @tinpham5614 @Robel-003 @Seiyaroo
> 
> I would prefer if this was done by @Robel-003 @tinpham5614 or @CourtneyHoppus as they have permission to use our issue naming convention for this and can create the correct issue to target this in a separate PR.

My mistake, I'll close this pull request then.

taylorpapke reviewed (approved):
All tests are passing for me just like in your screenshots. @CourtneyHoppus have all the changes that you requested been made?

CourtneyHoppus reviewed (changes requested):
Still reviewing this, but I can see some of the tests aren't quite right and/or complete.
I will continue to work on this.

heosman commented:
> @heosman please see the PR #71 I updated the testing for the delete function and all of the work you did on the archive function was based on that. One thing I changed was that when you check for the Bad Request Exception, you use an invalid id and the check in the function catches that automatically. Checking for the Not Found Exception should resolve to a null and we should still check that the model was called, as that is what the function is doing.

@CourtneyHoppus I looked over the changes in your pull request for the delete function test and implemented them in the archive function test. Let me know if it looks alright!

CourtneyHoppus commented:
Redirected to #73
